zingolib/src/lightclient/describe.rs

```diff
@@ -292,10 +289,12 @@ impl LightClient {
                     } else {
                         outgoing_tx_data.recipient_address.clone()
                     };
-                    // hash set is used to create unique list of addresses as duplicates are not inserted twice
-                    addresses.insert(address);
+                    // hash map is used to create unique list of addresses as duplicates are not inserted twice
+                    addresses.insert(address, outgoing_tx_data.output_index);
                 });
-            addresses.iter().for_each(|address| {
+            let mut addresses_vec = addresses.into_iter().collect::<Vec<_>>();
+            addresses_vec.sort_by_key(|x| x.1);
```

Comment:
now addresses are sorted by output_index. this is what caused the new blinking in the app. but why dont we sort value_transfers by, say timestamp? would this rustify work that is being done in zingo-mobile?

Reply:
I sorted already right before I read the results from the uniffi method... but doing this every time that the App is updating the wallet info is the actual problem of performance, IMO.
I believe that the same sorting in zingolib would be really fast and it make the performance of the consumers better, as well.
The sort that works for me is:
1. time
2. txid
3. address (I know can be empty)
4. pool (I know can be empty)

Reply:
> now addresses are sorted by output_index. this is what caused the new blinking in the app. but why dont we sort value_transfers by, say timestamp? would this rustify work that is being done in zingo-mobile?

Timestamp is the time the block was mined, and doesn't help with outputs in the same transaction. Output index is the order the outputs appear on-chain, which is the most canonical order, and it preserves pool order (all transparent outputs are first, then sapling, etc).

Reply:
If the only issue is with transactions with more than one value transfer, then output_index should make them fully deterministic..for any transactions received after this code lands. Oscar pointed me at some code that can rescan old transactions to fill in missing data, I need to enhance that code to make sure all old transactions are also ordered this way.